roomserver/storage/postgres/events_table.go

```diff
@@ -321,6 +327,9 @@ func (s *eventStatements) bulkSelectStateAtEventAndReference(
 		result.EventID = eventID
 		result.EventSHA256 = eventSHA256
 	}
+	if err = rows.Err(); err != nil {
```

Comment:
rows.Err() must be checked after the call to rows.Next(). So you might need to bump this code up a few lines.

Reply:
Not sure I understand what you mean. `rows.Err()` should be called ONLY after `rows.Next()` returns `false`. Thus in cases where we do  `for rows.Next() { .. }` the best place to call `rows.Err()` is right after the for loop.

Reply:
> right after the for loop

Yes, that is precisely what I thought.

Reply:
I still don't follow. This `rows.Err()` invoke is right after the very `for` looping over `rows.Next()`.

Reply:
Oh, you are right. It seemed to me that `rows.Scan` was outside the loop, my bad.
Ignore my comment.